backend/src/services/buildingService.js

```diff
@@ -0,0 +1,78 @@
+import { kingdomRepo } from '../repos/kingdomRepo';
+import { buildingRepo } from '../repos/buildingRepo';
+import { resourceService } from './resourceService';
+
+const errorMessages = {
+  missingType: 1,
+  wrongType: 2,
+  notEnoughGold: 3,
+  invalidId: 4
+};
+
+const defaultData = {
+  farm : { hp : 1, cost : 100, time : 60000, gen : 5, genType : 'food' },
+  mine : { hp : 1, cost : 100, time : 60000, gen : 5, genType : 'gold' },
+  academy : { hp : 1, cost : 100, time : 60000, gen : 0, genType : '-' }
+}
+
+const buildingData = (data) => {
```

Comment:
@gulyaasjaanos function names should be verbs usually, so `createBuildingData` maybe?